docroot/themes/contrib/civictheme/includes/utilities.inc

```diff
@@ -415,3 +420,110 @@ function civictheme_get_entity_field_key_values(EntityInterface $entity, $field_
 
   return $values;
 }
+
+/**
+ * Check if the provided link is external.
+ *
+ * @param string $url
+ *   The url to be validated.
+ * @param string $host
+ *   The current site host.
+ * @param array $overridden_domains
+ *   Array of overridden domains.
+ *
+ * @return bool
+ *   TRUE if the link is external, FALSE otherwise.
+ */
+function civictheme_link_is_external(string $url, string $host, array $overridden_domains) {
+  // If the link does not have a domain.
+  if ((strpos($url, 'internal:') === 0) || (strpos($url, '/') === 0)) {
+    return FALSE;
+  }
+  // If the link has a domain.
+  $url_data = parse_url($url);
+  if (!empty($url_data['host'])) {
+    $domain = trim($url_data['host'], '/');
+    // If domain is the same as the current site.
+    if ($domain == $host) {
+      return FALSE;
+    }
+    // Included in the list of the overridden domains, including wildcards.
+    if (!empty($overridden_domains)) {
+      foreach ($overridden_domains as $overridden_domain) {
+        if (strpos($overridden_domain, $domain) !== FALSE) {
+          return FALSE;
+        }
+      }
+    }
+  }
+  return TRUE;
+}
+
+/**
+ * Get the list of override domains.
+ *
+ * @return array
+ *   List of override domains.
+ */
+function civictheme_get_override_domains() {
+  $config_factory = \Drupal::service('config.factory');
+  $theme_name = $config_factory->get('system.theme')->get('default');
+  $config = $config_factory->get("$theme_name.settings");
+  $overridden_domains = $config->get('civictheme_override_domains') ?? '';
+  if (!empty($overridden_domains)) {
+    return explode("\r\n", $overridden_domains);
```

Comment:
please create a helper function and put into utlitlites to handle different new line characters

```
function multilineToArray($string) {
    $lines = is_array($string) ? $string : explode("\n", str_replace("\r\n", "\n", $string));
    return array_values(array_filter(array_map('trim', $lines)));
  }
```

Please copy the unit tests from here and add to the reset of unit tests

https://github.com/integratedexperts/testmode/blob/b193144ded607c33d83964313fa624faccccf65b/tests/src/Unit/TestmodeTest.php#L98